sourced/ml/mining/__main__.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import logging
+import sys
+
+from modelforge import slogging
+
+from .cmd import ArgumentDefaultsHelpFormatterNoNone
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """
+    Create the cmdline argument parser.
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=ArgumentDefaultsHelpFormatterNoNone
+    )
+    parser.add_argument(
+        "--log-level",
+        default="INFO",
+        choices=logging._nameToLevel,
+        help="Logging verbosity.",
+    )
+    # Create and construct subparsers
+    subparsers = parser.add_subparsers(help="Commands", dest="command")
+
+    def add_parser(name, help_message):
+        return subparsers.add_parser(
+            name, help=help_message, formatter_class=ArgumentDefaultsHelpFormatterNoNone
+        )
+
+    return parser
```

Comment:
```suggestion
    return parser.parse_args()
```

Reply:
Done

Reply:
actually I can't do this, or we don't get the parser in the main, which means we can't print help if needed

Reply:
can't do this or I can't display the help in the main